nasa-gibs/worldview
# WV-2919 notify fix

## Description
The recent Notification alert did not properly initialize the outage alert count in state, leading to an empty notification window. This fixes the state initialization to count actual OUTAGES, not layer notice outages.

Fixes # wv-2919

## How To Test
- git checkout wv-2919-notify-fix
- npm ci
- npm run watch

Test in normal mode with [this link](http://localhost:3000/)
- Confirm that on page load the notifications window pops up with OUTAGES listed only 
- Click Info Icon --> Notifications & confirm that all notifications are listed (not just the outages)

Test in EIC mode with [this link](http://localhost:3000/?eic=si)
- Confirm that on page load you are NOT shown any notifications
- Click Info Icon --> Notifications & confirm that all notifications are listed (not just the outages)
 
Test in KIOSK mode with [this link](http://localhost:3000/?kiosk=true)
- Confirm that on page load you are NOT shown any notifications
- Click Info Icon --> Notifications & confirm that all notifications are listed (not just the outages)

To simulate the Production environment, go to `web\js\containers\toolbar.js` & modify line 171 from
`requestNotifications(notificationURL);`
to
`requestNotifications('https://status.earthdata.nasa.gov/api/v1/notifications?domain=https://worldview.earthdata.nasa.gov');`

Test in normal mode with [this link](http://localhost:3000/)
- Confirm that the notification window does not appear (since there are no active alerts in production)

## Merging

Please use the `squash and merge` commit method unless each commit in your branch is vital to the commit history of main.

@nasa-gibs/worldview
